modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -234,6 +264,9 @@ Error GDScriptAnalyzer::resolve_inheritance(GDScriptParser::ClassNode *p_class,
 	}
 
 	GDScriptParser::DataType result;
+	result.kind = GDScriptParser::DataType::RESOLVING;
+	p_class->base_type = result;
+	result.kind = GDScriptParser::DataType::UNRESOLVED;
```

Comment:
Wouldn't line 268 invoke the copy constructor?

```
DataType(const DataType &p_other) {
	*this = p_other;
}
```
Looks like this would simply make `p_class->base_type` point to the same object as `result`, so that `p_class->base_type.kind` will be `UNRESOLVED`?

Reply:
yes, it copies the datatype over to `base_type` then switches the local variable `result`, but `base_type` is a copy, it isn't affected, i'll add a comment to this and others